.ci/visual-regression-test.sh

```diff
@@ -0,0 +1,127 @@
+    if ! GIT_FILE_MODIFIED 'composer.lock'
+    then
+        echo -e "\nSkipping visual regression tests since composer.lock has NOT changed"
+        exit 0
+    fi
+
+    # Skip visual tests if has been a modification to composer.json
+    if GIT_FILE_MODIFIED 'composer.json'
+    then
+        echo -e "\nSkipping visual regression tests since composer.json HAS changed"
+        exit 0
+    fi
+
+else
+    echo -e "\nRunning visual regression tests because the latest commit message demands it"
+fi
+
+# Ping the multidev environment to wake it from sleep
+echo -e "\nPinging the ${TERMINUS_ENV} multidev environment to wake it from sleep..."
+curl -I "$MULTIDEV_SITE_URL" >/dev/null
+
+# Ping the live environment to wake it from sleep
+echo -e "\nPinging the live environment to wake it from sleep..."
+curl -I "$LIVE_SITE_URL" >/dev/null
+
+# Check for custom backstop.json
+if [ ! -f ./.ci/backstop.json ]; then
+	# Create Backstop config file with dynamic URLs
+	echo -e "\nCreating backstop.js config file..."
+	node .ci/create-backstop-json-from-default.js
+fi
+
+# Backstop visual regression
+echo -e "\nRunning backstop reference..."
+
+echo -e "\nRunning backstop reference on ${LIVE_SITE_URL}..."
+backstop reference --config=.ci/backstop.json
+
+# Kill any zombie Chrome instances
+pkill -f "(chrome)?(--headless)"
+
+echo -e "\nRunning backstop test on ${MULTIDEV_SITE_URL}..."
+VISUAL_REGRESSION_RESULTS=$(backstop test --config=.ci/backstop.json || echo 'true')
+
+echo "${VISUAL_REGRESSION_RESULTS}"
+
+# Copy backstop_data files to ARTIFACTS_FULL_DIR
+echo -e "\nCopying backstop_data files to $ARTIFACTS_FULL_DIR..."
+rm -rf $ARTIFACTS_FULL_DIR/backstop_data
+cp -r backstop_data $ARTIFACTS_FULL_DIR/
+
+DIFF_IMAGE=$(find ./backstop_data -type f -name "*.png" | grep diff | grep desktop | head -n 1)
+
+if [ ! -f $DIFF_IMAGE ]; then
+	echo -e "\nDiff image file $DIFF_IMAGE not found!"
+fi
+
+DIFF_IMAGE_URL="$ARTIFACTS_DIR_URL/$DIFF_IMAGE"
+
+DIFF_REPORT="$ARTIFACTS_FULL_DIR/backstop_data/html_report/index.html"
+
+if [ ! -f $DIFF_REPORT ]; then
+	echo -e "\nDiff report file $DIFF_REPORT not found!"
+	exit 1
+fi
+
+DIFF_REPORT_URL="$ARTIFACTS_DIR_URL/backstop_data/html_report/index.html"
+
+REPORT_LINK="[![Visual report]($DIFF_IMAGE_URL)]($DIFF_REPORT_URL)"
+
+if [[ ${VISUAL_REGRESSION_RESULTS} == *"Mismatch errors found"* ]]
+then
+	# visual regression failed
+	echo -e "\nVisual regression test failed!"
+	PR_MESSAGE="Visual regression test failed! $REPORT_LINK"
+else
+	# visual regression passed
+	REPORT_LINK="\n\nView the [visual regression test report]($DIFF_REPORT_URL)"
+	echo -e "\n\nVisual regression test passed!"
+	PR_MESSAGE="Visual regression test passed! $REPORT_LINK"
+fi
+
+# Post the image back to the pull request on GitHub
+echo -e "\nPosting visual regression results back to PR #$PR_NUMBER "
+curl -s -i -u "$CI_PROJECT_USERNAME:$GITHUB_TOKEN" -d "{\"body\": \"$PR_MESSAGE\"}" $GITHUB_API_URL/issues/$PR_NUMBER/comments > /dev/null
```

Comment:
I guess you are using `curl` here rather than `terminus build:commit:comment:add` is that we're running in the backstop.js image, and Build Tools are not available.

Reply:
Correct